src/components/TrackingConsent.vue

```diff
@@ -0,0 +1,309 @@
+            >{{ text.no }}</button>
+            <button
+                class="nq-button-s"
+                @click="allowsBrowserData"
+                :class="{ inverse: theme === 'dark' }"
+            >{{ text.browserOnly }}</button>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
+
+interface Consents {
+    allowsBrowserData?: boolean;
+    allowsUsageData?: boolean;
+}
+
+@Component
+class TrackingConsent extends Vue {
+    @Prop({
+        type: Object,
+        default:  () => ({
+            main: 'Help Nimiq improve by sharing anonymized usage data. Thank you!',
+            yes: 'Yes',
+            no: 'No',
+            browserOnly: 'Browser-info only',
+        }),
+        validator: (value) => (
+            value && typeof value === 'object' &&
+            value.no && typeof value.no === 'string' && value.no.length &&
+            value.yes && typeof value.yes === 'string' && value.yes.length &&
+            value.main && typeof value.main === 'string' && value.main.length &&
+            value.browserOnly && typeof value.browserOnly === 'string' && value.browserOnly.length
+        ),
+    })
+    private text: {
+        main: string,
+        yes: string,
+        no: string,
+        browserOnly: string,
+    };
+
+    @Prop({
+        type: String,
+        default: 'light',
+        validator: (theme) => ['dark', 'light'].includes(theme),
+    })
+    private theme: string;
+
+    @Prop({
+        type: Object,
+        default: () => ({
+            setTrackerUrl: TrackingConsent.MATOMO_URL + 'nimiq.php',
+        }),
+    })
+    private options: {
+        setSiteId: string, // 3 for safe.nimiq.com ?
+        setTrackerUrl: string
+        addDownloadExtensions?: string,
+        trackPageView?: boolean,
+        enableLinkTracking?: boolean,
+        [k: string]: string | boolean,
+    };
+
+    @Prop({
+        type: String,
+        default: 'nimiq.js',
+    })
+    private tagManagerScript: string;
+
+    private consentKnown: boolean = false;
+    public static _storage: Consents;
+
+    private async mounted() {
+        if (!window.startTime) {
+            window.startTime = (new Date().getTime());
+        }
+
+        if (TrackingConsent.consents.allowsBrowserData || TrackingConsent.consents.allowsUsageData) {
+            this._initMatomo();
+        }
+
+        /* Automatically track stats if user is not in the EU continent  */
+        const geoIpResponse = await fetch(TrackingConsent.GEOIP_SERVER);
+        if (geoIpResponse.status !== 200) {
+            throw new Error('Failed to contact geoip server');
+        }
+        const geoIpInfo = await geoIpResponse.json();
+        if (geoIpInfo.continent !== 'EU') {
+            this.allowsConsent();
+        }
+    }
+
+    private denyConsent(): void {
+        this._setConsent({ allowsUsageData: false, allowsBrowserData: false });
```

Comment:
Would be good to also disable tracking here in case it was already active.
I guess `_paq.push(['optUserOut']);` as described [here](https://developer.matomo.org/guides/tracking-javascript-guide#optional-creating-a-custom-opt-out-form) is suitable for this.
`_paq.push(['forgetUserOptOut']);` would then be needed to be called when we want to enable tracking again.